pages/electronics-library.tsx

```diff
@@ -181,9 +190,22 @@ const Home = ({
     filterComponents();
   }, [searchCategoryTags, searchField]);
 
+  useEffect(() => {
+    const id = router.asPath.match(/(?<=#).*/g);
+    if (id) {
+      const queriedProduct = electronicComponents.find((component) =>
+        component.id === id[0]
+      );
+      queriedProduct && setActiveProduct(queriedProduct);
+      setShowOverlay(true);
+    }
+  }, [router])
+
   const metaTracker = useRef(false);
 
   useEffect(() => {
+    const changeHashQuery = (e: HashChangeEvent) => router.push(`/electronics-library${window.location.hash}`, undefined, { shallow: true })
```

Comment:
I assume this is for if the user changes the URL, but does not press `enter`?

Reply:
When user changes url manually and presses enter. Without it, manual changes after the hashbang wouldn't have effect

Reply:
Does `router.asPath` not catch that?

Reply:
I wasn't able to get it to